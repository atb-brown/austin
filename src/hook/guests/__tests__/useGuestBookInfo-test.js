import { registerMockProvider } from "../../../test/mock/hook/mockUseGuestBookInfo";
import TestComponentUseGuestBookInfo from "./TestComponentUseGuestBookInfo";
import { render, waitFor } from "@testing-library/react";

it("Basic Render", async () => {
  registerMockProvider();
  const renderResult = render(<TestComponentUseGuestBookInfo />);
  const fragment = renderResult.asFragment;

  // Wait for the underlying promise to resolve.
  await waitFor(() => renderResult.getByText("message: info loaded"));

  expect(fragment()).toMatchSnapshot();
});
