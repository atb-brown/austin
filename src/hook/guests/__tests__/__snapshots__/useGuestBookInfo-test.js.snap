// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Render 1`] = `
<DocumentFragment>
  <div>
    <div>
      guestCount: 1
    </div>
    <div>
      all guests: [{"name":"austin","message":"info loaded","visitDate":"2023","website":"www.example.com"}]
    </div>
    <div>
      name: austin
    </div>
    <div>
      message: info loaded
    </div>
    <div>
      visitDate: 2023
    </div>
    <div>
      website: www.example.com
    </div>
  </div>
</DocumentFragment>
`;
